controllers/asosecret_controller.go

```diff
@@ -300,25 +300,6 @@ func (asos *ASOSecretReconciler) createSecretFromClusterIdentity(ctx context.Con
 			newASOSecret.Data["AZURE_CLIENT_CERTIFICATE"] = identitySecret.Data["certificate"]
 			newASOSecret.Data["AZURE_CLIENT_CERTIFICATE_PASSWORD"] = identitySecret.Data["password"]
 		}
-	} else {
```

Comment:
should we remove the `if` altogether here?

Reply:
Removing the `if` caused a nil pointer dereference with the asosecret controller reconcile test if IdentityRef is not configured. Although we always expect IdentityRef to be populated at this point due to the validating webhook, would it still be safer to include this `if`, or just removing it along with the unit tests?

Test: https://github.com/willie-yao/cluster-api-provider-azure/blob/require-azureclusteridentity/controllers/asosecret_controller_test.go#L65

Reply:
I'm +1 for removing it since there is a validation in place to enforce it

If it's nil other things in the controller will fail later. We already assume other required things in this function aren't nil, like `cluster`